Patch by Sergei Golovan increases timeouts in S2S and removes horrible 5-minute
delay between remote server connection attempts after a falure (in case of
server is down it would be reasonable, but what happens much more often is a
network split when the delay is inadmissible).

--- ejabberd-2.0.0.orig/src/ejabberd_s2s_out.erl
+++ ejabberd-2.0.0/src/ejabberd_s2s_out.erl
@@ -90,7 +90,7 @@
 %% Only change this value if you now what your are doing:
 -define(FSMLIMITS,[]).
 %% -define(FSMLIMITS, [{max_queue, 2000}]).
--define(FSMTIMEOUT, 5000).
+-define(FSMTIMEOUT, 60000).
 
 -define(STREAM_HEADER,
 	"<?xml version='1.0'?>"
@@ -217,8 +217,8 @@
 	{error, _Reason} ->
 	    ?INFO_MSG("s2s connection: ~s -> ~s (remote server not found)",
 		      [StateData#state.myname, StateData#state.server]),
-	    wait_before_reconnect(StateData, 300000)
-	    %%{stop, normal, StateData}
+	    %%wait_before_reconnect(StateData, 300000)
+	    {stop, normal, StateData}
     end;
 open_socket(stop, StateData) ->
     ?INFO_MSG("s2s connection: ~s -> ~s (stopped in open socket)",
